internal/authentication/tripper.go

```diff
@@ -0,0 +1,29 @@
+package authentication
+
+import (
+	"fmt"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/types"
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+)
+
+var _ http.RoundTripper = (*TokenTripper)(nil)
+
+type TokenTripper struct {
+	Tripper     http.RoundTripper
+	TokenGetter *TokenGetter
+	Key         types.NamespacedName
+}
+
+func (tt *TokenTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	reqClone := utilnet.CloneRequest(req)
+	token, err := tt.TokenGetter.Get(reqClone.Context(), tt.Key)
+	if err != nil {
+		return nil, err
```

Comment:
One thing I think worth noting is the comment on the http.RoundTripper interface that calls out scenarios where RoundTrip errors:

>	// RoundTrip must always close the body, including on errors,
	// but depending on the implementation may do so in a separate
	// goroutine even after RoundTrip returns. This means that
	// callers wanting to reuse the body for subsequent requests
	// must arrange to wait for the Close call before doing so.
	
Should we still return the error here, or should we log the error and let the chain continue with an empty token value (that will likely result in installation failure)?

Reply:
Should be resolved by: https://github.com/operator-framework/operator-controller/pull/1074/files#diff-5afa0f97777f966f560e34685db2c58fe887216c09e41def816eb4d9232f45e3R25-R29